### PR TITLE
inet_db: fix a bug when .hosts file is never reloaded

### PR DESCRIPTION
**Pull request by max-au**
If .hosts file does not exist when kernel application starts, it will never get reloaded.

### CODE REVIEW

**Comment by IngelaAndin** [comment]
Could you please base your PR on maint

**Comment by RaimoNiskanen** [comment]
@max-au: Sorry about the delay!  I finally found some time to have a look at this.

There are internal options for `inet_db` named `hosts_file_name` and `resolv_conf_name` that should be used if you have not specified `hosts_file` and `resolv_conf`, which should be the default case, and that do exactly this, so I wonder why that does not work for you?

On the other hand the documentation http://erlang.org/doc/apps/erts/inet_cfg.html for `hosts_file` and `resolv_conf` states that they should be effective even if the files do not exist.  I think the documentation has crept away from the reality here.  Presumably, when the undocumented options were introduced, the intention was that the old options `hosts_file` and `resolv_conf` should work as since long, and if the `/etc/hosts` file (and `/etc/resolv.conf`) does not exist they are not supposed to exist and no info should be read from them.  But the default behaviour became to use the undocumented `hosts_file_name` and `resolv_conf_name` which should do exactly what you want.  Then someone changed the documentation of `hosts_file` and `resolv_conf` to describe the default behaviour, not these option's behaviour...

That said:  Why do you want to make this change?  Because if the default configuration does not read the `/etc/hosts` and `/etc/resolv.conf` file even if they do not exist at start there is a bug that needs to be fixed, but if you have a configuration with a custom Inet Config file that use the `hosts_file` and/or `resolv_conf` options that does not read the files when they appear later, the solution could be to use the `hosts_file_name` and `resolv_conf_name` options and that these should be documented, and that the documentation should be corrected for `hosts_file` and `resolv_conf`.  Or, maybe the `hosts_file` and `resolv_conf` options should be corrected according to their documentation, but that should be done on the `master` branch since it is a behaviour change...

File: lib/kernel/src/inet_db.erl [hunk 1]
```diff
@@ -1223,7 +1223,9 @@ handle_set_file(Option, Fname, TagTm, TagInfo, ParseFun, From,
 			    {ok, B, _} -> B;
 			    _ -> <<>>
 			end;
-		    _ -> <<>>
+		    _ ->
+			ets:insert(Db, {TagTm, times()}),
```
Comment (RaimoNiskanen) [inline lib/kernel/src/inet_db.erl]: This line should do what the clause for `ParseFun =:= undefined` does:
```
	ets:insert(Db, {TagInfo, undefined}),
	TimeZero = - (?RES_FILE_UPDATE_TM + 1), % Early enough
	ets:insert(Db, {TagTm, TimeZero}),
```
There are use cases where e.g `/etc/resolv.conf` file is written by the Erlang application immediately after system start, and by setting `{TagTm, times()}` a re-read is delayed for quite a while but by setting `{TagTm, TimeZero}` the file will be re-read at first resolver lookup.  Also, `TagInfo` is everywhere else in `inet_db` set in conjunction with `TagTm` so it feels more consistent to set it to `undefined` than to not set it.

**Comment by RaimoNiskanen** [comment]
I implemented my suggested review correction myself and merged that

**Comment by RaimoNiskanen** [comment]
I implemented my suggested review correction myself and merged that, after deciding to regard the current behaviour as a bug.

**Comment by max-au** [comment]
Sorry, missed the previous update.
Thank you Raimo, I also considered this as a bug, and happy it's fixed now. So I can file some more about inet_db :-)

**Pull request merged**